packages/bff/src/model/api/apiConverter.ts

```diff
@@ -0,0 +1,241 @@
+    },
+  };
+}
+
+export function toEserviceCatalogProcessQueryParams(
+  queryParams: BffGetCatalogApiQueryParam
+): CatalogProcessApiQueryParam {
+  return {
+    ...queryParams,
+    producersIds: queryParams.producersIds
+      ? queryParams.producersIds.join(",")
+      : undefined,
+    states: queryParams.states ? queryParams.states.join(",") : undefined,
+    attributesIds: queryParams.attributesIds
+      ? queryParams.attributesIds.join(",")
+      : undefined,
+    agreementStates: queryParams.agreementStates
+      ? queryParams.agreementStates.join(",")
+      : undefined,
+  };
+}
+
+export function toBffCatalogApiEServiceResponse(
+  eservice: CatalogProcessApiEService,
+  producerTenant: TenantProcessApiTenant,
+  hasCertifiedAttributes: boolean,
+  isRequesterEqProducer: boolean,
+  activeDescriptor?: CatalogProcessApiEServiceDescriptor,
+  agreement?: AgreementProcessApiAgreement
+): BffCatalogApiEServiceResponse {
+  const isUpgradable = (agreement: AgreementProcessApiAgreement): boolean => {
+    const eserviceDescriptor = eservice.descriptors.find(
+      (e) => e.id === agreement.descriptorId
+    );
+
+    return (
+      eserviceDescriptor !== undefined &&
+      eservice.descriptors
+        .filter((d) => Number(d.version) > Number(eserviceDescriptor.version))
+        .find(
+          (d) =>
+            (d.state === descriptorApiState.PUBLISHED ||
+              d.state === descriptorApiState.SUSPENDED) &&
+            (agreement.state === agreementApiState.ACTIVE ||
+              agreement.state === agreementApiState.SUSPENDED)
+        ) !== undefined
+    );
+  };
+
+  const partialEnhancedEservice = {
+    id: eservice.id,
+    name: eservice.name,
+    description: eservice.description,
+    producer: {
+      id: eservice.producerId,
+      name: producerTenant.name,
+    },
+    isMine: isRequesterEqProducer,
+    hasCertifiedAttributes,
+  };
+
+  return {
+    ...partialEnhancedEservice,
+    ...(activeDescriptor
+      ? {
+          activeDescriptor: {
+            id: activeDescriptor.id,
+            version: activeDescriptor.version,
+            audience: activeDescriptor.audience,
+            state: activeDescriptor.state,
+          },
+        }
+      : {}),
+    ...(agreement
+      ? {
+          agreement: {
+            id: agreement.id,
+            state: agreement.state,
+            canBeUpgraded: isUpgradable(agreement),
+          },
+        }
+      : {}),
+  };
+}
+
+export function toDescriptorState(
+  state: CatalogProcessApiEServiceDescriptorState
+): DescriptorState {
+  return match(state)
+    .with("DRAFT", () => descriptorState.draft)
+    .with("PUBLISHED", () => descriptorState.published)
+    .with("DEPRECATED", () => descriptorState.deprecated)
+    .with("ARCHIVED", () => descriptorState.archived)
+    .with("SUSPENDED", () => descriptorState.suspended)
+    .exhaustive();
+}
+
+export function toEServiceDescriptorDocument(
```

Comment:
it seems this function is not used. Can you tell me where this was supposed to be used?

Reply:
Right again this is a wrong conflict, fixed [here](https://github.com/pagopa/interop-be-monorepo/pull/665/commits/1edd1de6c620fa5a15957e036a93f4c3f06d81f6)